dev_docs/07_visual_testing_guide.md

```diff
@@ -0,0 +1,93 @@
+# Visual Testing
+
+The KDS now has a visual testing system integrated into it, using which you can verify the expected visual states after making any changes to the components.
+
+## Implementation details
```

Comment:
I wonder if this section should be the last one in this file. I think one could be more interesting in reading first what it is and how to use it, and then how was this implemented (as probably less people would be interested in this last point).

Reply:
But this section contains the details of all components. One should be aware of how things are working in the background, what all packages are being used, and the use of all the utility functions before writing tests. I'm okay with both the points though. 
Do let me know what you think :)

Reply:
Im okay with both points too :), It was just a thought, if you consider ok to have this section in this position go ahead!

Reply:
Yes, I am pretty sure that as users of the visual testing system, developers won't really dig, and shouldn't need to, into the implementation details. So it's really better located at the end.

I think it's great to have it though! I am sure that it will help a lot any time we need to maintain the system itself.

Reply:
> and the use of all the utility functions before writing tests

That's a good point @KshitijThareja. I'd suggest we actually move "how to use" parts from the second section to the first one so that could help. In the second section, only "how it works" would stay.